source/extensions/filters/http/geoip/geoip_filter.h

```diff
@@ -25,6 +26,11 @@ class GeoipFilterConfig {
   bool useXff() const { return use_xff_; }
   uint32_t xffNumTrustedHops() const { return xff_num_trusted_hops_; }
 
+  // Returns true if a custom header should be used to extract the IP address.
+  bool useIpAddressHeader() const { return ip_address_header_.has_value(); }
+  // Returns the header name to use for extracting the IP address.
+  const Http::LowerCaseString& ipAddressHeader() const { return ip_address_header_.value(); }
```

Comment:
what do we return here if `ip_address_header_` is nullopt?

Reply:
could be better to return optional than ref to string

Reply:
Okay, let me change it to return `absl::optional` instead.